pkg/manifests/manifests_test.go

```diff
@@ -3622,6 +3622,38 @@ enableUserWorkload: true
 	}
 }
 
+func TestPrometheusOperatorConfigurationHyperShift(t *testing.T) {
```

Comment:
Not mandatory but a single test function with table-driven test cases would be neat :)
We could also check that setups without hosted control plane preserve the node selector.